hcatalog/server-extensions/src/main/java/org/apache/hive/hcatalog/listener/DbNotificationListener.java

```diff
@@ -858,19 +859,65 @@ private void addNotificationLog(NotificationEvent event, ListenerEvent listenerE
       long nextNLId = getNextNLId(stmt, sqlGenerator,
               "org.apache.hadoop.hive.metastore.model.MNotificationLog");
 
-      String insertVal = "(" + nextNLId + "," + nextEventId + "," + now() + ", ?, ?," +
-              quoteString(" ") + ",?, ?)";
+      String insertVal;
+      String columns;
+      List<String> params = new ArrayList<String>();
+
+      // Construct the values string, parameters and column string step by step simultaneously so
+      // that the positions of columns and of their corresponding values do not go out of sync.
+
+      // Notification log id
+      columns = "\"NL_ID\"";
```

Comment:
Can combine all compulsory columns in one line instead of concatenating individually.

Reply:
There are 6 compulsory arguments. All of those do not fit in a line. Some of those are passed as parameters and some as values to the insert statement. One needs to look at the values, count commas in order to match column names with values or parameters. Building them one at a time makes it very clear as to which column name goes with which value and/or parameter. So, I thought this little refactoring would improve readability a lot. Do you see any problem with this arrangement?